server/server.go

```diff
@@ -1248,7 +1248,7 @@ func (s *Server) Start() {
 	// own system account if one is not present.
 	if opts.JetStream {
 		var cfg *JetStreamConfig
-		if opts.StoreDir != "" {
+		if opts.StoreDir != "" || opts.JetStreamMaxMemory != 0 || opts.JetStreamMaxStore != 0 {
```

Comment:
I think there's some related problems here

https://github.com/nats-io/nats-server/blob/9ac7dfd9abab32eb49ddaddac825ca78a1c4a78a/server/jetstream.go#L102-L109

this checks only these settings and not if storedir is empty and then it goes and set storedir only if config is nil, not if config is nil or storedir is empty - this resolves later but its quite convoluted

maybe here we should set cfg to `dynJetStreamConfig()` and then set these values if they are set, this way all the munging and deciding whats in config is in one place